iina/PrefKeyBindingViewController.swift

```diff
@@ -341,66 +263,103 @@ class PrefKeyBindingViewController: NSViewController, PreferenceWindowEmbeddable
 
   // MARK: - Private
 
-  private func loadConfigFile() {
-    if let mapping = KeyMapping.parseInputConf(at: currentConfFilePath) {
-      mappingController.content = nil
-      mappingController.add(contentsOf: mapping)
-      mappingController.setSelectionIndexes(IndexSet())
-    } else {
-      // on error
+  /// This function firstly reloads the table data, select the config file row, then load the config file.
+  /// If the target config file cannot be found, or the file cannot be parsed correctly, it will fallback to the default config.
+  /// - Parameter configName: the target config name
+  private func loadConfigFile(_ configName: String?) {
+    guard configName != Preference.string(for: .currentInputConfigName) else { return }
+    isLoading = true
+    
+    func fallback() {
+      isLoading = false
       Utility.showAlert("keybinding_config.error", arguments: [currentConfName], sheetWindow: view.window)
-      let title = "IINA Default"
-      currentConfName = title
-      currentConfFilePath = getFilePath(forConfig: title)!
-      confTableSelectRow(withTitle: title)
-      loadConfigFile()
-      return
+      loadConfigFile(fallbackDefault)
     }
+
+    guard let configName = configName else { fallback(); return }
+    
+    cachedConfigNames = configNames
+    confTableView.reloadData()
+    if let index = cachedConfigNames.firstIndex(of: configName) {
+      confTableView.selectRowIndexes(IndexSet(integer: index), byExtendingSelection: false)
+    }
+    currentConfName = configName
+    currentConfFilePath = getFilePath(forConfig: configName)!
+    
```

Comment:
If you move around the last few lines like this, you can get rid of the `isLoading` variable completely:

```Swift

    guard let configName = configName else { fallback(); return }
    
    cachedConfigNames = configNames
    currentConfName = configName
    currentConfFilePath = getFilePath(forConfig: configName)!

    confTableView.reloadData()
    if let index = cachedConfigNames.firstIndex(of: configName) {
      confTableView.selectRowIndexes(IndexSet(integer: index), byExtendingSelection: false)
    }

```

Reply:
Thanks for your suggestion. I too, indeed want to get rid of `isLoading`, however, I tried, it caused more edge cases which will make the code messier. It's not worth it to explain how it would cause problems which requires more code to deal with, but trust me here :)

Reply:
Would be good to at least change the name of the variable to something more specific (e.g., "isLoadingConf" and maybe copy your comment above about infinite loops into a code comment to make it easier to figure out its purpose.